layers/icmp4.go

```diff
@@ -231,17 +231,28 @@ func (i *ICMPv4) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error
 	return nil
 }
 
+// fix from google/gopacket pull request by https://github.com/cjacobs96
+// https://github.com/google/gopacket/pull/754
 // SerializeTo writes the serialized form of this layer into the
 // SerializationBuffer, implementing gopacket.SerializableLayer.
 // See the docs for gopacket.SerializableLayer for more info.
 func (i *ICMPv4) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
-	bytes, err := b.PrependBytes(8)
+	//bytes, err := b.PrependBytes(8)
+	bytes, err := b.PrependBytes(8 + len(i.Payload)) //header size + payload size
 	if err != nil {
 		return err
 	}
 	i.TypeCode.SerializeTo(bytes)
 	binary.BigEndian.PutUint16(bytes[4:], i.Id)
 	binary.BigEndian.PutUint16(bytes[6:], i.Seq)
+
+	startIndex := 8 //start at after header offset
+	//copy payload into buffer
+	for _, element := range i.Payload {
```

Comment:
This is incorrect.
According to [BaseLayer godoc](https://pkg.go.dev/github.com/gopacket/gopacket@v1.3.0/layers#BaseLayer), `i.Payload` is **the set of bytes contained by (but not part of) this Layer**.
If you want to encode payload in ICMPv4 packet, it should be a separate `gopacket.Payload` layer.

Reply:
ok, I guess that makes sense. can this reconcile with the fact that not including a payload produces truncated ICMPv4 ping? I feel like the default behavior should still be a complete ICMPv4, but maybe you mean the logic in constructing the packet should always include Payload layer separately and we're doing it wrong to achieve non-truncated result as users of the library?

Reply:
> the logic in constructing the packet should always include Payload layer separately

Yes, if you need the payload.